### ISSUE TITLE
Build fails if git repo hasn't yet been initialized for project

### ISSUE THREAD

**Issue by edeandrea** [n0mer/gradle-git-properties#35]
As part of an organization we've modified the Spring Initializr (https://start.spring.io) so that it generates a cradle-based project which includes the gradle-git-properties plugin pre-wired into the build. The issue we're having is that once someone creates a project they may not have checked their code into a git repo yet. Not doing that - they are unable to build their project:

```
$ ./gradlew build
Build cache is an incubating feature.
Using directory (/Users/user/.gradle/caches/build-cache-1) as local build cache, push is enabled.
:bootBuildInfo UP-TO-DATE
:compileJava UP-TO-DATE
:generateGitProperties FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':generateGitProperties'.
> org.eclipse.jgit.errors.RepositoryNotFoundException: repository not found: /Users/user/Desktop/mycompany-spring-starter

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
```

I can put in a workaround by adding this to the `build.gradle`:

```
generateGitProperties {
	onlyIf {
		!source.isEmpty()
	}
}
```

I think if you added the [@SkipWhenEmpty](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/SkipWhenEmpty.html) annotation on top of the `@InputFiles` annotation on `GenerateGitPropertiesTask.getSource()` method inside `GitPropertiesPlugin` then all consumers of your plugin would simply get it for free.

**Comment by mjustin** [comment]
Isn't this a duplicate of https://github.com/n0mer/gradle-git-properties/issues/27?

**Comment by edeandrea** [comment]
Ahh yes I believe it might be.

**Comment by n0mer** [comment]
fixed in 1.4.18
